### DISCUSSION TITLE
Unable to handle uncaught errors in production mode

### DISCUSSION THREAD

**Discussion by aerialist7** [remix-run/react-router#10534] [Q&A]
Hi Team!

I use a global error handler to catch exceptions.

I found that these exceptions are caught only in development mode but not in production when I use react-router.

Here is an example - https://github.com/aerialist7/unhandled-in-prod-react-router

0. Run `npm install`
1. Run app (`npm run start` for development mode or `npm run build` + `npm run serve` for production mode)
2. Open the browser console
3. Go to the "Error" route 
4. You will find a log from the global error handler in development mode but there will be no log in production :(

Why it works this way? 
Is it a bug?

P.S. When I removed react-router usage - exceptions are caught both in development and production modes

**Comment by aerialist7** [comment]
Any news?

**Comment by aerialist7** [comment]
@brophdawg11, @timdorr, hi! 
Sorry for the inconvenience, but there is no answer for a month...

May be I should file an issue?

**Comment by brophdawg11** [comment]
I don't think this is an issue with React Router, but instead due to something going on in `react-scripts`.  When I run this app in via `vite` I get the global error logged in both development and production mode.  You can also add your own `ErrorBoundary` and throw or log the error in there.

You may want to keep an eye on this proposal as well which is looking for a way to let errors bubble outside the router: https://github.com/remix-run/react-router/discussions/10166
